.clang-tidy

```diff
@@ -0,0 +1,74 @@
+---
```

Comment:
TODO for me: check that the clang-tidy stuff works out of the box on our development machines.

Reply:
I have a grand plan of applying `clang-tidy -fix` to the codebase, probably after enforcing clang-format.

It is pretty slow though since, I think, it reanalyzes headers for each compilation unit (approximately 45minutes for the entire code base: https://github.com/KatanaGraph/katana/runs/566296025).

My IDE just picks up this configuration so I've been applying the linter as I go for new files.

Reply:
Sounds great. Let's try to do this after more of the new code for this release gets merged/finalized.

Reply:
We have a clang-tidy executable already built with our existing clang and the existing script works, so no further intervention from me is needed for packaging there.